### PR TITLE
Can create and can update false reason

### PR DESCRIPTION
**Pull request by jrchudy**
This PR addresses issue #677 . The following text will be added before the error text (`You are not authorized to perform this action. Please report this problem to your system administrators.`) when the user navigates to `recordedit` for a table they can't create or update on:

Text reasons for `canCreate`:
```
  case view:
     "Table is a view."
  case generated:
     "Table is generated."
  case no insert permission for user:
     "No permissions to create."
  case all columns disabled:
     "All columns are disabled."
```

Text reasons for `canUpdate`:
```
  case view:
     "Table is a view."
  case generated:
     "Table is generated."
  case no insert permission for user:
     "No permissions to update."
  case all columns disabled:
     "All columns are disabled."
```

### CODE REVIEW

File: js/reference.js [hunk 1]
```diff
@@ -877,18 +906,50 @@
...
+        /**
+         * Indicates the reason as to why a user cannot update for
+         * the referenced resource(s). In some cases, this won't be set
+         * because the user can update, so the value will be `undefined`.
+         * @type {(String|undefined)}
+         */
+        get canUpdateReason() {
+            if (this._canUpdateReason === undefined) {
+                // will set the _canUpdateReason property
+                this.canUpdate();
```
Comment (RFSH) [inline js/reference.js]: canUpdate is not a function, so you cannot call it that way. You have to say `var res = this.canUpdate;`

File: js/reference.js [hunk 2]
```diff
@@ -836,18 +836,47 @@
...
+        /**
+         * Indicates the reason as to why a user cannot create for
+         * the referenced resource(s). In some cases, this won't be set
+         * because the user can create, so the value will be `undefined`.
+         * @type {(String|undefined)}
+         */
+        get canCreateReason() {
+            if (this._canCreateReason === undefined) {
+                // will set the _canCreateReason property
+                this.canCreate();
```
Comment (RFSH) [inline js/reference.js]: Same comment as `canUpdateReason`, `canCreate` is not a function.

File: js/reference.js [hunk 3]
```diff
@@ -836,18 +836,47 @@
                 // 3) not all visible columns in the table are generated
                 var ref = (this._context === module._contexts.CREATE) ? this : this.contextualize.entryCreate;
 
-                this._canCreate = ref._table.kind !== module._tableKinds.VIEW && !ref._table._isGenerated && ref._checkPermissions("insert");
+                if (ref._table.kind === module._tableKinds.VIEW) {
+                    this._canCreate = false;
+                    this._canCreateReason = "Table is a view. ";
```
Comment (RFSH) [inline js/reference.js]: Can you please add an object like the ones that we have [at the bottom of utilities.js](https://github.com/informatics-isi-edu/ermrestjs/blob/master/js/utilities.js#L3182) to have these messages in one place?

**Comment by hongsudt** [comment]
Instead of putting those sentences at the beginning, it might be better to put them in the detail section.

**Pull request merged**